bin/main.rb

```diff
@@ -1,118 +1,159 @@
 
 while player2.empty?
-  puts 'Second player name: '
+  puts "\t Invalid name, try again!
+  \n Second player name: "
   player2 = gets.chomp.capitalize
 end
 
 game_trial = Players.new(player1, player2)
 
-puts "\n #{game_trial.player1} is going to play 'X', and #{game_trial.player2} will play 'O'"
-puts "Let's start"
+puts "\n \t #{game_trial.player1} is going to play 'X', and #{game_trial.player2} will play 'O'"
+puts "\t Let's start"
 
 sleep(3)
 
 game_trial.start
 
+sleep(2)
+
+# Combinations to win
+combinations_likely_to_win = [[1, 2, 3], [2, 5, 8], [3, 6, 9], [4, 5, 6], [1, 4, 7], [1, 5, 9], [3, 5, 7], [7, 8, 9]]
+
+# Create permutations
+combinations_likely_to_win.each do |values_to_win|
+  combinations_likely_to_win += [values_to_win.rotate(2)]
+  combinations_likely_to_win += [values_to_win.rotate(-2)]
+end
+
 # Variable can control the iterations
-number_turn = 0
-game_turn = true
+iterator = 0
+winner = 0
+game_state = true
+move_repeated = false
 
 # To Provide turns for each one player
-players_turn = lambda {
+player1_turn = lambda {
   system 'clear'
 
+  next unless winner.zero?
+
   game_trial.display_board
-  puts "It's #{game_trial.player1}'s turn \n"
-  puts "Reminder: You're 'X"
-  puts 'Please select a a number between 1 to 9 according to the board.'
+  puts "\t It's #{game_trial.player1}'s turn \n"
+  puts "\t Reminder: You're 'X'"
   x_selected = gets.chomp.to_i
-  until (1..9).include? x_selected
-    puts 'Invalid, please select a number between 1 to 9.'
+  move_repeated = game_trial.check_moves_repeated(x_selected)
+  until ((1..9).include? x_selected) && !move_repeated
+    puts "\t Invalid, please select a number between 1 to 9 and don' repeat them."
     x_selected = gets.chomp.to_i
+    move_repeated = game_trial.check_moves_repeated(x_selected)
   end
+  game_trial.save_marker_player1(x_selected)
+  game_trial.update_board(x_selected, 'X')
+  game_trial.moves_used(x_selected)
+  system 'clear'
+  game_trial.display_board
+  sleep(1)
+}
 
+player2_turn = lambda {
   system 'clear'
+  next unless winner.zero?
 
   game_trial.display_board
-  puts "It's #{game_trial.player2}'s turn \n"
-  puts "Reminder: You're 'O"
-  puts 'Please select a a number between 1 to 9 according to the board.'
+  puts "\t It's #{game_trial.player2}'s turn \n"
+  puts "\t Reminder: You're 'O'"
   o_selected = gets.chomp.to_i
-  until (1..9).include? o_selected
-    puts 'Invalid, please select a number between 1 to 9.'
+  move_repeated = game_trial.check_moves_repeated(o_selected)
+  until ((1..9).include? o_selected) && !move_repeated
+    puts "\t Invalid, please select a number between 1 to 9 and don' repeat them."
     o_selected = gets.chomp.to_i
+    move_repeated = game_trial.check_moves_repeated(o_selected)
+  end
+  game_trial.save_marker_player2(o_selected)
+  game_trial.update_board(o_selected, 'O')
+  game_trial.moves_used(o_selected)
+  system 'clear'
+  game_trial.display_board
+  sleep(1)
+}
```

Comment:
- [ ] You should follow the DRY rule. Instead of repeating each part for each player. Please extract repeated parts to a method and call the method for each player. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))